src/app/pages/checkout-payment/payment-payone-ideal/payment-payone-ideal.component.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { Attribute } from 'ish-core/models/attribute/attribute.model';
+import { PaymentMethod } from 'ish-core/models/payment-method/payment-method.model';
+import { markAsDirtyRecursive } from 'ish-shared/forms/utils/form-utils';
+
+@Component({
+  selector: 'ish-payment-payone-ideal',
+  templateUrl: './payment-payone-ideal.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PaymentPayoneIdealComponent implements OnInit {
+  constructor(protected cd: ChangeDetectorRef) {}
+
+  /**
+   * payone payment method, needed to get configuration parameters
+   */
+  @Input() paymentMethod: PaymentMethod;
+
+  /**
+   * should be set to true by the parent, if component is visible
+   */
+  @Input() activated = false;
+
+  @Output() cancel = new EventEmitter<void>();
+  @Output() submit = new EventEmitter<{ parameters: Attribute<string>[]; saveAllowed: boolean }>();
```

Comment:
events should not be named as standard DOM events - our eslint integration is coming soon and will mark this as an error, e.g. submitPayment would be fine
find more information here: https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin/src/rules/no-output-native.ts

Reply:
I tried changing the names, but the submit and cancel buttons stopped responding.